File: tests/fixtures/negative/android-lint/HandlerLeak.java
package com.example;

import android.os.Handler;
import android.os.Looper;
import android.os.Message;

class MyActivity {
    static class StaticHandler extends Handler {
        @Override
        public void handleMessage(Message msg) {
            // handle safely
        }
    }

    class LooperHandler extends Handler {
        LooperHandler(Looper looper) {
            super(looper);
        }

        @Override
        public void handleMessage(Message msg) {
            // handle safely
        }
    }
}

File: tests/fixtures/negative/android-lint/HandlerLeak.kt
package com.example

import android.os.Handler
import android.os.Looper

class MyActivity {
    // Not an inner class, so no implicit reference to outer
    class SafeHandler(looper: Looper) : Handler(looper) {
        override fun handleMessage(msg: android.os.Message) {
            // handle safely
        }
    }

    inner class LooperHandler(looper: Looper) : Handler(looper) {
        override fun handleMessage(msg: android.os.Message) {
            // handle safely
        }
    }
}
